src/test/operators/join_test_runner.cpp

```diff
@@ -180,20 +197,92 @@ class JoinTestRunner : public BaseTestWithParam<JoinTestConfiguration> {
       PredicateCondition::Equals,
       all_secondary_predicate_sets.front(),
       std::make_shared<JoinOperatorFactory<JoinOperator>>(),
+      std::nullopt,
       std::nullopt
     };
     // clang-format on
 
-    const auto add_configuration_if_supported = [&](const auto& configuration) {
-      // String vs non-String comparisons are not supported in Hyrise and therefore cannot be tested
-      if ((configuration.data_type_left == DataType::String) != (configuration.data_type_right == DataType::String)) {
-        return;
+    /**
+     * Returns a set of adapted configurations if the join type provides further configuration possibilities that
+     * should be tested. Otherwise, a vector containing only the passed configuration is returned.
+     */
+    const auto build_join_type_specific_variations = [&](const auto& configuration) {
+      if constexpr (std::is_same_v<JoinOperator, JoinIndex>) {
+        // Currently the Index Join cannot deal with configurations where the join column
+        // data types are different, see Issue #2077
+        if (configuration.data_type_left != configuration.data_type_right) {
+          return std::vector<JoinTestConfiguration>{};
+        }
+
+        // For the JoinIndex, additional parameters influence its execution behavior:
+        //  - index side (left or right)
+        //  - availability of an index for a join column segment
+        //  - presence of "single chunk reference guarantee" for join column reference segments
+
+        // share of indexed segments; 1 means all segments are indexed
+        std::array<float, 2> indexed_segment_shares{1.0f, .1f};
```

Comment:
is .1f meaningful for most configurations (i.e., do most configs have at least 10 chunks)?

Reply:
Could be improved with `std::ceil`?

Before:
`indexed_chunk_count = static_cast<uint32_t>(indexed_share * chunk_count);`

After:
`indexed_chunk_count = static_cast<uint32_t>(std::ceil(indexed_share * chunk_count))`

Whith this extension, we have at least one indexed segment.

Reply:
Sounds good! Add a check that this is not the same as chunk_count, otherwise skip it?

Reply:
In the current implementation, we add the calculated `indexed_chunk_range` for each `indexed_share` to a set of `ChunkRange`s. If the calculated range for share `.1f` would cover all segments, it wouldn't be a problem here since we also have `1.0f`, i.e., we want to have the test case in which all segments are indexed. Since a set is used for storing the ranges, we don't have duplicates of `ChunkRange`s.

```
auto indexed_chunk_ranges = std::set<ChunkRange>{};
for (const auto indexed_share : indexed_segment_shares) {
            indexed_chunk_ranges.insert(ChunkRange{0, static_cast<uint32_t>(std::ceil(indexed_share * chunk_count))});
          }

          for (const auto& indexed_chunk_range : indexed_chunk_ranges) {
             [...]
```